keyboards/treasure/type30/hotswap/keyboard.json

```diff
@@ -0,0 +1,121 @@
+{
+    "manufacturer": "TREASURE",
+    "keyboard_name": "TYPE30",
+    "maintainer": "TREASURE",
+    "bootloader": "atmel-dfu",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "extrakey": true,
+        "nkro": true,
+        "rgb_matrix": true
+    },
+
+    "matrix_pins": {
+        "cols": ["B5", "B6", "B7", "C7", "C6", "C4", "D4", "D5", "D6", "B0", "B1", "B2"],
+        "rows": ["D2", "C5", "B3"]
+    },
+    "processor": "atmega32u2",
+    "rgb_matrix": {
+        "animations": {
+            "gradient_up_down": true,
+            "gradient_left_right": true
+        },
+        "driver": "ws2812",
+        "default": {
+            "speed": 40,
+            "animation": "gradient_left_right"
+        },
+        "layout": [
```

Comment:
These coordinates do not appear to be correct. Per the RGB Matrix documentation:
> The default expected range of values for `{ x, y }` is the inclusive range `{ 0..224, 0..64 }`.

Reply:
Ok updated

Reply:
And sorry for the disruption , I reopened the PR with a much much simpler firmware so it’s cleaner and easier